multicorp-bot/Helpers/TelemetryHelper.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+using DSharpPlus.CommandsNext;
+using DSharpPlus.Entities;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.DependencyCollector;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+
+namespace multicorp_bot.Helpers
+{
+    
+    /// <summary>
+    /// A class used to log telemety into Azure Application Insights.  It creates pretty charts and can log most things.
+    /// This class is provisioned in the startup as a singleton object, then reference throughout the app.
+    /// </summary>
+    public class TelemetryHelper
+    {
+        private const string DEFAULT_INSTRUMENT_KEY = "f3552a4b-4698-4742-9ec9-044fd132d10a";
+        private TelemetryConfiguration _config = null;
+        private TelemetryClient _client = null;
+        private string _environment = string.Empty;
+        private static readonly TelemetryHelper _helper = new TelemetryHelper();
+        public static TelemetryHelper Singleton => _helper;
+
+        public TelemetryHelper()
+        {
+            try
+            {
+                // Gets the key from the environment variables.
+                string key = Environment.GetEnvironmentVariable("TelemetryAppKey");
+                if (string.IsNullOrEmpty(key))
+                {
+                    key = DEFAULT_INSTRUMENT_KEY;
+                }
+
+                // Create the telemetry configuration
+                TelemetryConfiguration configuration = TelemetryConfiguration.CreateDefault();
+                configuration.InstrumentationKey = key;
+                this._config = configuration;
+
+                // Create the environment property we'll read later, to correlate logs.
+                string env = Environment.GetEnvironmentVariable("TelemetryAppKey");
+                if (string.IsNullOrEmpty(env))
+                {
+                    env = "DEV";
+                }
+                this._environment = Environment.GetEnvironmentVariable("TelemetryENV");
```

Comment:
Technically this will likely crash with a null reference if the environment variable is not added. I forgot to update this.